docs/blueprints/referral-manager-v1.md

```diff
@@ -43,61 +43,47 @@ Redeemed Satellite
+        - If the `users` table contains a value larger than `0` for `new_tokens` for this user ID, the Referral Manager will generate that number of new tokens, add the new unredeemed tokens to the `tokens` table, and set the value of `new_tokens` for that user to `0` in the `users` table.
+        - If there are unredeemed tokens in the `tokens` table, the Referral Manager will respond to the satellite's request with the tokens. Otherwise, it will return an empty response.
     - The satellite receives the response from Referral Manager.
         - If the response payload contains tokens, the satellite will display them in the UI.
         - If it's an empty response payload, the satellite will display a message `No available referral links. Try again later.`
 
 
 _Referral Manager CLI_
 
-1. `referral-manager start` initiates the invitation token generation process. This process begins by checking whether there are existing `eligibleUsers` in memory in Referral Manager.
-    - If `eligibleUsers` is empty, the Referral Manager queries `users` table to get users with 0 remaining invites.
-    - If `eligibleUsers` is not empty, it means one of two things:
-        * Someone else is using the Referral Manager at the same time and has not generated/sent tokens yet.
-        * The CLI was terminated between the get `eligibleUsers` step and the generate/send tokens step.
-    
-        Either one of the cases, We will return an error and display `Referral token generation already in progress or canceled early. Do you want to continue anyway?`. If they decide to continue, we call `ClearUsers` to set `eligibleUsers` to empty and the Referral Manager queries the `users` table to get users with 0 remaining invites. 
-2. The Referral Manager returns the number of users for each satellite to the CLI.
-3. The CLI receives the number of users and displays: `How many invitations do you want to generate for each user?`
-4. After the operator enters the number of invites per user to generate, the CLI displays: `Generating X invitation tokens for Y users on Z satellites. Yes/No.`
-   - If the input is `Yes`, it will generate the amount of invitation tokens per user based on the input from CLI and save them into `tokens` table.
-   - The CLI will call `ClearUsers` endpoint to set `eligibleUser` to be empty if the input is `No`.
+1. `referral-manager start --tokens-per-user=3 <satelliteURL1> <satelliteURL2> ...` initiates the invitation token generation process. The `--dry-run` flag can optionally be added to run the process without actually generating tokens. The dry run allows for estimating how many new tokens would be created by the command.
+2. The Referral Manager queries the `users` table for all users where `new_tokens < flags.tokens_per_user` and where the satellite matches one of the CLI-provided satellite URLs. These are the "eligible users".
```

Comment:
I think here we will need to also look into the `tokens` table since we only want to generate tokens for users whose unredeemed token is less than x amount. Maybe it would be something like `new_tokens + unredeemed_tokens < flags.eligible_user_token_count` ? I think `flags.tokens_per_user` is the input for the amount of new tokens we want to create for each `eligibleUser`